compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -608,73 +608,6 @@ impl<'tcx> Body<'tcx> {
         self.injection_phase.is_some()
     }
 
-    /// If this basic block ends with a [`TerminatorKind::SwitchInt`] for which we can evaluate the
-    /// dimscriminant in monomorphization, we return the discriminant bits and the
-    /// [`SwitchTargets`], just so the caller doesn't also have to match on the terminator.
-    fn try_const_mono_switchint<'a>(
-        tcx: TyCtxt<'tcx>,
-        instance: Instance<'tcx>,
-        block: &'a BasicBlockData<'tcx>,
-    ) -> Option<(u128, &'a SwitchTargets)> {
-        // There are two places here we need to evaluate a constant.
-        let eval_mono_const = |constant: &ConstOperand<'tcx>| {
-            let env = ty::ParamEnv::reveal_all();
-            let mono_literal = instance.instantiate_mir_and_normalize_erasing_regions(
-                tcx,
-                env,
-                crate::ty::EarlyBinder::bind(constant.const_),
-            );
-            mono_literal.try_eval_bits(tcx, env)
-        };
```

Comment:
This cannot be handled by ordinary MIR opts because many times constants or branches on constants cannot be evaluated in a polymorphic MIR opt.

Reply:
Also FYI attempting to remove the mono-reachable traversal entirely will break builds of compiler-builtins, and should cause the run-make test for this case to fail.

Reply:
> This cannot be handled by ordinary MIR opts because many times constants or branches on constants cannot be evaluated in a polymorphic MIR opt.

can this be removed once/if your post-monomorphization MIR opts PR lands?

Reply:
Yup. That's one of the goals of that PR.